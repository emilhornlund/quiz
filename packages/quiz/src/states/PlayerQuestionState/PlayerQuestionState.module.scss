@import '../../styles/colors';
@import '../../styles/helpers';

.main {
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  flex: 1;

  @include mobile {
    gap: $layout-gutter;
    padding: $layout-gutter * 0.5 0;
  }

  @include tablet {
    gap: $layout-gutter * 1.5;
    padding: $layout-gutter * 1 0;
  }

  @include desktop {
    gap: $layout-gutter * 2;
    padding: $layout-gutter * 1.5 0;
  }

  .question {
    font-weight: bold;
    text-align: center;

    @include mobile {
      @include fontSize(16px);
    }

    @include tablet {
      @include fontSize(46px);
    }

    @include desktop {
      @include fontSize(56px);
    }
  }

  .answer {
    display: flex;
    flex-direction: column;
    flex: 1;
  }

  .timer {
    display: flex;
    flex-direction: column;
    background-color: $white-1-50;

    span {
      display: flex;
      flex-direction: column;
      height: 100%;
      background-color: $white;
    }

    @mixin timer-dimensions($height) {
      height: pxToRem($height);
      border-radius: pxToRem($height / 2);

      span {
        border-radius: pxToRem($height / 2);
      }
    }

    @include mobile {
      @include timer-dimensions(4px);
    }

    @include tablet {
      @include timer-dimensions(6px);
    }

    @include desktop {
      @include timer-dimensions(8px);
    }
  }
}
