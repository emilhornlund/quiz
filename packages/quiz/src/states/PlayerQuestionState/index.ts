export type { PlayerQuestionStateProps } from './PlayerQuestionState'
export { default } from './PlayerQuestionState'
