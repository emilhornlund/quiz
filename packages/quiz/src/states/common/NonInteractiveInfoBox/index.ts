export type { NonInteractiveInfoBoxProps } from './NonInteractiveInfoBox'
export { default } from './NonInteractiveInfoBox'
