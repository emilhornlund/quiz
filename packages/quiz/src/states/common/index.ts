export type { NonInteractiveInfoBoxProps } from './NonInteractiveInfoBox'
export type { PointBehindIndicatorProps } from './PointsBehindIndicator'
export type { QuestionAnswerPickerProps } from './QuestionAnswerPicker'
export type { QuestionMediaProps } from './QuestionMedia'
export type { QuestionTextPreviewProps } from './QuestionTextPreview'
export type { QuestionTypePointsBarProps } from './QuestionTypePointsBar'

export { default as GamePage } from './GamePage'
export { default as NonInteractiveInfoBox } from './NonInteractiveInfoBox'
export { default as PointsBehindIndicator } from './PointsBehindIndicator'
export { default as QuestionAnswerPicker } from './QuestionAnswerPicker'
export { default as QuestionMedia } from './QuestionMedia'
export { default as QuestionTextPreview } from './QuestionTextPreview'
export { default as QuestionTypePointsBar } from './QuestionTypePointsBar'
