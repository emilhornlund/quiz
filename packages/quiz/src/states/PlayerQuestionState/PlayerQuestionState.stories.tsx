import { GameEventType, MediaType, QuestionType } from '@quiz/common'
import type { Meta, StoryObj } from '@storybook/react'
import { withRouter } from 'storybook-addon-remix-react-router'

import PlayerQuestionState from './PlayerQuestionState'

const meta = {
  component: PlayerQuestionState,
  decorators: [withRouter],
  parameters: {
    layout: 'fullscreen',
  },
} satisfies Meta<typeof PlayerQuestionState>

export default meta
type Story = StoryObj<typeof meta>

export const QuestionMultiChoiceTwoAnswers = {
  args: {
    event: {
      type: GameEventType.GameQuestionPlayer,
      player: {
        nickname: 'FrostyBear',
        score: { total: 10458 },
      },
      question: {
        type: QuestionType.MultiChoice,
        question: 'Who painted The Starry Night?',
        media: {
          type: MediaType.Image,
          url: 'https://i.pinimg.com/originals/a6/60/72/a66072b0e88258f2898a76c3f3c01041.jpg',
        },
        answers: [{ value: 'Vincent van Gogh' }, { value: 'Pablo Picasso' }],
        duration: 30,
      },
      countdown: {
        expiryTime: new Date(Date.now() + 60 * 1000).toISOString(),
        serverTime: new Date().toISOString(),
      },
      pagination: { current: 1, total: 20 },
    },
    onSubmitQuestionAnswer: () => undefined,
  },
} satisfies Story

export const QuestionMultiChoiceFourAnswers = {
  args: {
    event: {
      type: GameEventType.GameQuestionPlayer,
      player: {
        nickname: 'FrostyBear',
        score: { total: 10458 },
      },
      question: {
        type: QuestionType.MultiChoice,
        question: 'Who painted The Starry Night?',
        media: {
          type: MediaType.Image,
          url: 'https://wallpapercave.com/wp/wp2824407.jpg',
        },
        answers: [
          { value: 'Vincent van Gogh' },
          { value: 'Pablo Picasso' },
          { value: 'Leonardo da Vinci' },
          { value: 'Claude Monet' },
        ],
        duration: 30,
      },
      countdown: {
        expiryTime: new Date(Date.now() + 60 * 1000).toISOString(),
        serverTime: new Date().toISOString(),
      },
      pagination: { current: 1, total: 20 },
    },
    onSubmitQuestionAnswer: () => undefined,
  },
} satisfies Story

export const QuestionMultiChoiceSixAnswers = {
  args: {
    event: {
      type: GameEventType.GameQuestionPlayer,
      player: {
        nickname: 'FrostyBear',
        score: { total: 10458 },
      },
      question: {
        type: QuestionType.MultiChoice,
        question: 'Who painted The Starry Night?',
        media: {
          type: MediaType.Image,
          url: 'https://wallpaperaccess.com/full/157316.jpg',
        },
        answers: [
          { value: 'Vincent van Gogh' },
          { value: 'Pablo Picasso' },
          { value: 'Leonardo da Vinci' },
          { value: 'Claude Monet' },
          { value: 'Michelangelo' },
          { value: 'Rembrandt' },
        ],
        duration: 30,
      },
      countdown: {
        expiryTime: new Date(Date.now() + 60 * 1000).toISOString(),
        serverTime: new Date().toISOString(),
      },
      pagination: { current: 1, total: 20 },
    },
    onSubmitQuestionAnswer: () => undefined,
  },
} satisfies Story

export const QuestionSlider = {
  args: {
    event: {
      type: GameEventType.GameQuestionPlayer,
      player: {
        nickname: 'FrostyBear',
        score: { total: 10458 },
      },
      question: {
        type: QuestionType.Range,
        question: "What percentage of the earth's surface is covered by water?",
        min: 0,
        max: 100,
        step: 1,
        duration: 30,
      },
      countdown: {
        expiryTime: new Date(Date.now() + 60 * 1000).toISOString(),
        serverTime: new Date().toISOString(),
      },
      pagination: { current: 1, total: 20 },
    },
    onSubmitQuestionAnswer: () => undefined,
  },
} satisfies Story

export const QuestionTrueFalse = {
  args: {
    event: {
      type: GameEventType.GameQuestionPlayer,
      player: {
        nickname: 'FrostyBear',
        score: { total: 10458 },
      },
      question: {
        type: QuestionType.TrueFalse,
        question: "Rabbits can't vomit?",
        duration: 30,
      },
      countdown: {
        expiryTime: new Date(Date.now() + 60 * 1000).toISOString(),
        serverTime: new Date().toISOString(),
      },
      pagination: { current: 1, total: 20 },
    },
    onSubmitQuestionAnswer: () => undefined,
  },
} satisfies Story

export const QuestionTypeAnswer = {
  args: {
    event: {
      type: GameEventType.GameQuestionPlayer,
      player: {
        nickname: 'FrostyBear',
        score: { total: 10458 },
      },
      question: {
        type: QuestionType.TypeAnswer,
        question: 'Who painted the Mono Lisa?',
        duration: 30,
      },
      countdown: {
        expiryTime: new Date(Date.now() + 60 * 1000).toISOString(),
        serverTime: new Date().toISOString(),
      },
      pagination: { current: 1, total: 20 },
    },
    onSubmitQuestionAnswer: () => undefined,
  },
} satisfies Story
