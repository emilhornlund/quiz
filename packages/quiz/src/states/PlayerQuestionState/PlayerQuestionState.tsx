import { GameEventQuestionPlayer } from '@quiz/common'
import React, { FC } from 'react'

import QuestionAnswerPicker, { Page, PlayerGameFooter } from '../../components'

import styles from './PlayerQuestionState.module.scss'

export interface PlayerQuestionStateProps {
  event: GameEventQuestionPlayer
}

const PlayerQuestionState: FC<PlayerQuestionStateProps> = ({
  event: {
    nickname,
    question,
    time,
    score: { total: totalScore },
    pagination: { current: currentQuestion, total: totalQuestions },
  },
}) => (
  <Page
    footer={
      <PlayerGameFooter
        currentQuestion={currentQuestion}
        totalQuestions={totalQuestions}
        nickname={nickname}
        totalScore={totalScore}
      />
    }>
    <div className={styles.main}>
      <div className={styles.question}>{question.question}</div>
      <div className={styles.timer}>
        <span style={{ width: `${(time / question.duration) * 100}%` }} />
      </div>
      <QuestionAnswerPicker question={question} />
    </div>
  </Page>
)

export default PlayerQuestionState
