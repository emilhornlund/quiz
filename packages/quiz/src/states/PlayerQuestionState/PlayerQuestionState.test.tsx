import { GameEventType, MediaType, QuestionType } from '@quiz/common'
import { fireEvent, render, screen } from '@testing-library/react'
import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest'

const h = vi.hoisted(() => ({
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  submitQuestionAnswer: vi.fn<[], Promise<void>>().mockResolvedValue(),
}))

vi.mock('../../context/game', () => ({
  useGameContext: () => ({
    submitQuestionAnswer: h.submitQuestionAnswer,
  }),
}))

import PlayerQuestionState from './PlayerQuestionState'

const now = Date.now()

describe('PlayerQuestionState', () => {
  beforeEach(() => {
    vi.useFakeTimers()
    vi.setSystemTime(new Date(now))
    h.submitQuestionAnswer.mockClear()
  })

  afterEach(() => {
    vi.useRealTimers()
  })

  it('should render PlayerQuestionState with question type multi-choice and two answers', () => {
    const { container } = render(
      <MemoryRouter>
        <PlayerQuestionState
          event={{
            type: GameEventType.GameQuestionPlayer,
            player: { nickname: 'FrostyBear', score: { total: 10458 } },
            question: {
              type: QuestionType.MultiChoice,
              question: 'Who painted The Starry Night?',
              media: {
                type: MediaType.Image,
                url: 'https://i.pinimg.com/originals/a6/60/72/a66072b0e88258f2898a76c3f3c01041.jpg',
              },
              answers: [
                { value: 'Vincent van Gogh' },
                { value: 'Pablo Picasso' },
              ],
              duration: 30,
            },
            countdown: {
              initiatedTime: new Date(now).toISOString(),
              expiryTime: new Date(now + 60 * 1000).toISOString(),
              serverTime: new Date(now).toISOString(),
            },
            pagination: { current: 1, total: 20 },
          }}
        />
      </MemoryRouter>,
    )
    expect(container).toMatchSnapshot()
  })

  it('should render PlayerQuestionState with question type multi-choice and four answers', () => {
    const { container } = render(
      <MemoryRouter>
        <PlayerQuestionState
          event={{
            type: GameEventType.GameQuestionPlayer,
            player: { nickname: 'FrostyBear', score: { total: 10458 } },
            question: {
              type: QuestionType.MultiChoice,
              question: 'Who painted The Starry Night?',
              media: {
                type: MediaType.Image,
                url: 'https://wallpapercave.com/wp/wp2824407.jpg',
              },
              answers: [
                { value: 'Vincent van Gogh' },
                { value: 'Pablo Picasso' },
                { value: 'Leonardo da Vinci' },
                { value: 'Claude Monet' },
              ],
              duration: 30,
            },
            countdown: {
              initiatedTime: new Date(now).toISOString(),
              expiryTime: new Date(now + 60 * 1000).toISOString(),
              serverTime: new Date(now).toISOString(),
            },
            pagination: { current: 1, total: 20 },
          }}
        />
      </MemoryRouter>,
    )
    expect(container).toMatchSnapshot()
  })

  it('should render PlayerQuestionState with question type multi-choice and six answers', () => {
    const { container } = render(
      <MemoryRouter>
        <PlayerQuestionState
          event={{
            type: GameEventType.GameQuestionPlayer,
            player: { nickname: 'FrostyBear', score: { total: 10458 } },
            question: {
              type: QuestionType.MultiChoice,
              question: 'Who painted The Starry Night?',
              media: {
                type: MediaType.Image,
                url: 'https://wallpaperaccess.com/full/157316.jpg',
              },
              answers: [
                { value: 'Vincent van Gogh' },
                { value: 'Pablo Picasso' },
                { value: 'Leonardo da Vinci' },
                { value: 'Claude Monet' },
                { value: 'Michelangelo' },
                { value: 'Rembrandt' },
              ],
              duration: 30,
            },
            countdown: {
              initiatedTime: new Date(now).toISOString(),
              expiryTime: new Date(now + 60 * 1000).toISOString(),
              serverTime: new Date(now).toISOString(),
            },
            pagination: { current: 1, total: 20 },
          }}
        />
      </MemoryRouter>,
    )
    expect(container).toMatchSnapshot()
  })

  it('should render PlayerQuestionState with question type range', () => {
    const { container } = render(
      <MemoryRouter>
        <PlayerQuestionState
          event={{
            type: GameEventType.GameQuestionPlayer,
            player: { nickname: 'FrostyBear', score: { total: 10458 } },
            question: {
              type: QuestionType.Range,
              question:
                "What percentage of the earth's surface is covered by water?",
              min: 0,
              max: 100,
              step: 1,
              duration: 30,
            },
            countdown: {
              initiatedTime: new Date(now).toISOString(),
              expiryTime: new Date(now + 60 * 1000).toISOString(),
              serverTime: new Date(now).toISOString(),
            },
            pagination: { current: 1, total: 20 },
          }}
        />
      </MemoryRouter>,
    )
    expect(container).toMatchSnapshot()
  })

  it('should render PlayerQuestionState with question type true false', () => {
    const { container } = render(
      <MemoryRouter>
        <PlayerQuestionState
          event={{
            type: GameEventType.GameQuestionPlayer,
            player: { nickname: 'FrostyBear', score: { total: 10458 } },
            question: {
              type: QuestionType.TrueFalse,
              question: "Rabbits can't vomit?",
              duration: 30,
            },
            countdown: {
              initiatedTime: new Date(now).toISOString(),
              expiryTime: new Date(now + 60 * 1000).toISOString(),
              serverTime: new Date(now).toISOString(),
            },
            pagination: { current: 1, total: 20 },
          }}
        />
      </MemoryRouter>,
    )
    expect(container).toMatchSnapshot()
  })

  it('should render PlayerQuestionState with question type type answer', () => {
    const { container } = render(
      <MemoryRouter>
        <PlayerQuestionState
          event={{
            type: GameEventType.GameQuestionPlayer,
            player: { nickname: 'FrostyBear', score: { total: 10458 } },
            question: {
              type: QuestionType.TypeAnswer,
              question: 'Who painted the Mono Lisa?',
              duration: 30,
            },
            countdown: {
              initiatedTime: new Date(now).toISOString(),
              expiryTime: new Date(now + 60 * 1000).toISOString(),
              serverTime: new Date(now).toISOString(),
            },
            pagination: { current: 1, total: 20 },
          }}
        />
      </MemoryRouter>,
    )
    expect(container).toMatchSnapshot()
  })

  it('submits answer when clicking an option (multi-choice)', async () => {
    render(
      <MemoryRouter>
        <PlayerQuestionState
          event={{
            type: GameEventType.GameQuestionPlayer,
            player: { nickname: 'FrostyBear', score: { total: 10458 } },
            question: {
              type: QuestionType.MultiChoice,
              question: 'Who painted The Starry Night?',
              answers: [
                { value: 'Vincent van Gogh' },
                { value: 'Pablo Picasso' },
              ],
              duration: 30,
            },
            countdown: {
              initiatedTime: new Date(now).toISOString(),
              expiryTime: new Date(now + 60 * 1000).toISOString(),
              serverTime: new Date(now).toISOString(),
            },
            pagination: { current: 1, total: 20 },
          }}
        />
      </MemoryRouter>,
    )
    fireEvent.click(screen.getByText('Vincent van Gogh'))
    expect(h.submitQuestionAnswer).toHaveBeenCalledTimes(1)
  })

  it('does not throw when submitQuestionAnswer is undefined', () => {
    // @ts-expect-error undefined
    h.submitQuestionAnswer = undefined
    render(
      <MemoryRouter>
        <PlayerQuestionState
          event={{
            type: GameEventType.GameQuestionPlayer,
            player: { nickname: 'FrostyBear', score: { total: 10458 } },
            question: {
              type: QuestionType.MultiChoice,
              question: 'Pick one',
              answers: [{ value: 'A' }, { value: 'B' }],
              duration: 30,
            },
            countdown: {
              initiatedTime: new Date(now).toISOString(),
              expiryTime: new Date(now + 60 * 1000).toISOString(),
              serverTime: new Date(now).toISOString(),
            },
            pagination: { current: 1, total: 20 },
          }}
        />
      </MemoryRouter>,
    )
    fireEvent.click(screen.getByText('A'))
  })
})
