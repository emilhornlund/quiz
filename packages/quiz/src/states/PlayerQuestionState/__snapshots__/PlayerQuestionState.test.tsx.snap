// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayerQuestionState > should render PlayerQuestionState with question type multi and four answers 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <div
        class="logo"
      >
        <span
          class="icon"
        />
        <span
          class="text"
        >
          Quiz
        </span>
      </div>
      <div
        class="side"
      />
    </div>
    <div
      class="content center"
    >
      <div
        class="main"
      >
        <div
          class="question"
        >
          Who painted The Starry Night?
        </div>
        <div
          class="timer"
        >
          <span
            style="width: 66.66666666666666%;"
          />
        </div>
        <div
          class="main"
        >
          <div
            class="main"
          >
            <div
              class="grid"
            >
              <button
                class="item"
                id="0_Vincent van Gogh"
                type="button"
              >
                <div />
                Vincent van Gogh
              </button>
              <button
                class="item"
                id="1_Pablo Picasso"
                type="button"
              >
                <div />
                Pablo Picasso
              </button>
              <button
                class="item"
                id="2_Leonardo da Vinci"
                type="button"
              >
                <div />
                Leonardo da Vinci
              </button>
              <button
                class="item"
                id="3_Claude Monet"
                type="button"
              >
                <div />
                Claude Monet
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="footer"
    >
      <div
        class="main"
      >
        <div
          class="questions"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-question "
            data-icon="circle-question"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM169.8 165.3c7.9-22.3 29.1-37.3 52.8-37.3l58.3 0c34.9 0 63.1 28.3 63.1 63.1c0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6c-13.3 0-24-10.7-24-24l0-13.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1c0-8.4-6.8-15.1-15.1-15.1l-58.3 0c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"
              fill="currentColor"
            />
          </svg>
          <span>
            1
             / 
            20
          </span>
        </div>
        <div
          class="nickname"
        >
          <span>
            FrostyBear
          </span>
        </div>
        <div
          class="score"
        >
          <span>
            10458
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PlayerQuestionState > should render PlayerQuestionState with question type multi and six answers 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <div
        class="logo"
      >
        <span
          class="icon"
        />
        <span
          class="text"
        >
          Quiz
        </span>
      </div>
      <div
        class="side"
      />
    </div>
    <div
      class="content center"
    >
      <div
        class="main"
      >
        <div
          class="question"
        >
          Who painted The Starry Night?
        </div>
        <div
          class="timer"
        >
          <span
            style="width: 66.66666666666666%;"
          />
        </div>
        <div
          class="main"
        >
          <div
            class="main"
          >
            <div
              class="grid"
            >
              <button
                class="item"
                id="0_Vincent van Gogh"
                type="button"
              >
                <div />
                Vincent van Gogh
              </button>
              <button
                class="item"
                id="1_Pablo Picasso"
                type="button"
              >
                <div />
                Pablo Picasso
              </button>
              <button
                class="item"
                id="2_Leonardo da Vinci"
                type="button"
              >
                <div />
                Leonardo da Vinci
              </button>
              <button
                class="item"
                id="3_Claude Monet"
                type="button"
              >
                <div />
                Claude Monet
              </button>
              <button
                class="item"
                id="4_Michelangelo"
                type="button"
              >
                <div />
                Michelangelo
              </button>
              <button
                class="item"
                id="5_Rembrandt"
                type="button"
              >
                <div />
                Rembrandt
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="footer"
    >
      <div
        class="main"
      >
        <div
          class="questions"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-question "
            data-icon="circle-question"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM169.8 165.3c7.9-22.3 29.1-37.3 52.8-37.3l58.3 0c34.9 0 63.1 28.3 63.1 63.1c0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6c-13.3 0-24-10.7-24-24l0-13.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1c0-8.4-6.8-15.1-15.1-15.1l-58.3 0c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"
              fill="currentColor"
            />
          </svg>
          <span>
            1
             / 
            20
          </span>
        </div>
        <div
          class="nickname"
        >
          <span>
            FrostyBear
          </span>
        </div>
        <div
          class="score"
        >
          <span>
            10458
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PlayerQuestionState > should render PlayerQuestionState with question type multi and two answers 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <div
        class="logo"
      >
        <span
          class="icon"
        />
        <span
          class="text"
        >
          Quiz
        </span>
      </div>
      <div
        class="side"
      />
    </div>
    <div
      class="content center"
    >
      <div
        class="main"
      >
        <div
          class="question"
        >
          Who painted The Starry Night?
        </div>
        <div
          class="timer"
        >
          <span
            style="width: 66.66666666666666%;"
          />
        </div>
        <div
          class="main"
        >
          <div
            class="main"
          >
            <div
              class="grid"
            >
              <button
                class="item"
                id="0_Vincent van Gogh"
                type="button"
              >
                <div />
                Vincent van Gogh
              </button>
              <button
                class="item"
                id="1_Pablo Picasso"
                type="button"
              >
                <div />
                Pablo Picasso
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="footer"
    >
      <div
        class="main"
      >
        <div
          class="questions"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-question "
            data-icon="circle-question"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM169.8 165.3c7.9-22.3 29.1-37.3 52.8-37.3l58.3 0c34.9 0 63.1 28.3 63.1 63.1c0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6c-13.3 0-24-10.7-24-24l0-13.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1c0-8.4-6.8-15.1-15.1-15.1l-58.3 0c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"
              fill="currentColor"
            />
          </svg>
          <span>
            1
             / 
            20
          </span>
        </div>
        <div
          class="nickname"
        >
          <span>
            FrostyBear
          </span>
        </div>
        <div
          class="score"
        >
          <span>
            10458
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PlayerQuestionState > should render PlayerQuestionState with question type slider 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <div
        class="logo"
      >
        <span
          class="icon"
        />
        <span
          class="text"
        >
          Quiz
        </span>
      </div>
      <div
        class="side"
      />
    </div>
    <div
      class="content center"
    >
      <div
        class="main"
      >
        <div
          class="question"
        >
          What percentage of the earth's surface is covered by water?
        </div>
        <div
          class="timer"
        >
          <span
            style="width: 66.66666666666666%;"
          />
        </div>
        <div
          class="main"
        >
          <div
            class="main"
          >
            <div
              class="interactive"
            >
              <form>
                <input
                  id="slider-range"
                  max="100"
                  min="0"
                  step="1"
                  type="range"
                  value="0"
                />
                <div
                  class="inputGroup"
                >
                  <div
                    class="main"
                  >
                    <input
                      data-testid="test-slider-input-textfield"
                      id="slider-input"
                      max="100"
                      min="0"
                      name="slider-input"
                      type="number"
                      value="0"
                    />
                  </div>
                  <div
                    class="main"
                  >
                    <button
                      class="primary"
                      data-testid="test-submit-button-button"
                      id="submit-button"
                      name="submit-button"
                      type="submit"
                    >
                      <span>
                        Submit
                      </span>
                    </button>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="footer"
    >
      <div
        class="main"
      >
        <div
          class="questions"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-question "
            data-icon="circle-question"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM169.8 165.3c7.9-22.3 29.1-37.3 52.8-37.3l58.3 0c34.9 0 63.1 28.3 63.1 63.1c0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6c-13.3 0-24-10.7-24-24l0-13.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1c0-8.4-6.8-15.1-15.1-15.1l-58.3 0c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"
              fill="currentColor"
            />
          </svg>
          <span>
            1
             / 
            20
          </span>
        </div>
        <div
          class="nickname"
        >
          <span>
            FrostyBear
          </span>
        </div>
        <div
          class="score"
        >
          <span>
            10458
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PlayerQuestionState > should render PlayerQuestionState with question type true false 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <div
        class="logo"
      >
        <span
          class="icon"
        />
        <span
          class="text"
        >
          Quiz
        </span>
      </div>
      <div
        class="side"
      />
    </div>
    <div
      class="content center"
    >
      <div
        class="main"
      >
        <div
          class="question"
        >
          Rabbits can't vomit?
        </div>
        <div
          class="timer"
        >
          <span
            style="width: 66.66666666666666%;"
          />
        </div>
        <div
          class="main"
        >
          <div
            class="main"
          >
            <div
              class="grid"
            >
              <button
                class="item"
                id="0_True"
                type="button"
              >
                <div />
                True
              </button>
              <button
                class="item"
                id="1_False"
                type="button"
              >
                <div />
                False
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="footer"
    >
      <div
        class="main"
      >
        <div
          class="questions"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-question "
            data-icon="circle-question"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM169.8 165.3c7.9-22.3 29.1-37.3 52.8-37.3l58.3 0c34.9 0 63.1 28.3 63.1 63.1c0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6c-13.3 0-24-10.7-24-24l0-13.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1c0-8.4-6.8-15.1-15.1-15.1l-58.3 0c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"
              fill="currentColor"
            />
          </svg>
          <span>
            1
             / 
            20
          </span>
        </div>
        <div
          class="nickname"
        >
          <span>
            FrostyBear
          </span>
        </div>
        <div
          class="score"
        >
          <span>
            10458
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PlayerQuestionState > should render PlayerQuestionState with question type type answer 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <div
        class="logo"
      >
        <span
          class="icon"
        />
        <span
          class="text"
        >
          Quiz
        </span>
      </div>
      <div
        class="side"
      />
    </div>
    <div
      class="content center"
    >
      <div
        class="main"
      >
        <div
          class="question"
        >
          Who painted the Mono Lisa?
        </div>
        <div
          class="timer"
        >
          <span
            style="width: 66.66666666666666%;"
          />
        </div>
        <div
          class="main"
        >
          <div
            class="main"
          >
            <div
              class="interactive"
            >
              <form>
                <div
                  class="main"
                >
                  <input
                    data-testid="test-answer-input-textfield"
                    id="answer-input"
                    name="answer-input"
                    placeholder="Answer"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="main"
                >
                  <button
                    class="primary"
                    data-testid="test-submit-button-button"
                    id="submit-button"
                    name="submit-button"
                    type="submit"
                  >
                    <span>
                      Submit
                    </span>
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="footer"
    >
      <div
        class="main"
      >
        <div
          class="questions"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-question "
            data-icon="circle-question"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM169.8 165.3c7.9-22.3 29.1-37.3 52.8-37.3l58.3 0c34.9 0 63.1 28.3 63.1 63.1c0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6c-13.3 0-24-10.7-24-24l0-13.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1c0-8.4-6.8-15.1-15.1-15.1l-58.3 0c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"
              fill="currentColor"
            />
          </svg>
          <span>
            1
             / 
            20
          </span>
        </div>
        <div
          class="nickname"
        >
          <span>
            FrostyBear
          </span>
        </div>
        <div
          class="score"
        >
          <span>
            10458
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
